src/core/load-fragment.c

```diff
@@ -3630,7 +3630,10 @@ int config_parse_allowed_cpus(
 
         CGroupContext *c = data;
 
-        (void) parse_cpu_set_extend(rvalue, &c->cpuset_cpus, true, unit, filename, line, lvalue);
+        if (streq(lvalue, "AllowedCPUs"))
+                (void) parse_cpu_set_extend(rvalue, &c->cpuset_cpus, true, unit, filename, line, lvalue);
+        else if (streq(lvalue, "StartupAllowedCPUs"))
+                (void) parse_cpu_set_extend(rvalue, &c->startup_cpuset_cpus, true, unit, filename, line, lvalue);
 
         return 0;
```

Comment:
i think we should merge config_parse_allowed_cpus() and config_parse_allowed_mems() into one, and then change the `data` parameter to get a pointer to `cgroup_context.cpuset_cpus` instead of `cgroup_context`, and siilar for the other fields.

That would shorten the code quite a bit, and make these lvalue string comparisons unnecessary.

Reply:
Yep, sounds much better. Done.